### PR TITLE
Update the release actions to use new cibuildwheel

### PR DESCRIPTION
**Pull request by mcflugen**
### Description

This pull request updates our CI to use the new *cibuildwheel* to build and publish *Landlab* release on TestPyPI and PyPI.

### Checklist - did you ...

- [x] Add a [news fragment](https://landlab.readthedocs.io/en/master/development/contribution/index.html#news-entries) file entry if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?
- [x] All tests have passed?
- [x] Formatted code with black?
- [x] Removed lint reported by flake8?
- [x] Sucessful documentation built? (if documentation added or modified)

### CODE REVIEW

**Review by mdpiper — APPROVED** [review]
This is fascinating. So many wheels to build! The TestPyPI and PyPI workflows are unlike anything I've seen before. There's even a graph on GitHub Actions!

I learned about *cibuildwheel* and *pipx* from this PR.

I like that you point to the docs for install instructions more complex than `pip install landlab`.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -59,7 +60,7 @@ jobs:
         env:
           HYPOTHESIS_PROFILE: "ci"
         run: |
-          nox -s test
+          nox -s test -- -m "${{ matrix.pytest-marker }}"
 
       - name: Coveralls
```
[the remark this answers is not in the document]
Comment (mcflugen) [reply .github/workflows/test.yml:65]: Good question. I've added the `parallel` flag to the *coveralls-python-action* and added a `coveralls_finish` job. In theory that should do the trick but we shall see.

[the remark this answers is not in the document]
Comment (mcflugen) [reply .github/workflows/test.yml:65]: Yup, that looks like it [did the trick](https://coveralls.io/builds/56756732).

**Comment by mcflugen** [comment]
> I like that you point to the docs for install instructions more complex than `pip install landlab`.

I did this initially because *PyPI* only allows descriptions written in only the most basic restructured text and we used some non-standard directives (i.e. `.. tab::`). In the end, though, I think this is probably better as the README's easier to read and it redirects users to our documentation.

**Review by mdpiper — APPROVED** [review]
👍

**Pull request merged**